app/Transformers/UserTransformer.php

```diff
@@ -5,96 +5,104 @@
 
 namespace App\Transformers;
 
+use App\Models\Beatmap;
 use App\Models\User;
 use League\Fractal;
 
-class UserTransformer extends TransformerAbstract
+class UserTransformer extends UserCompactTransformer
 {
-    protected $availableIncludes = [
-        'account_history',
-        'active_tournament_banner',
-        'badges',
-        'blocks',
-        'defaultStatistics',
-        'favourite_beatmapset_count',
-        'follower_count',
-        'friends',
-        'graveyard_beatmapset_count',
-        'group_badge',
-        'is_admin',
-        'loved_beatmapset_count',
-        'monthly_playcounts',
-        'page',
-        'previous_usernames',
-        'ranked_and_approved_beatmapset_count',
-        'replays_watched_counts',
-        'scores_first_count',
-        'statistics',
-        'support_level',
-        'unranked_beatmapset_count',
-        'unread_pm_count',
-        'user_achievements',
-        'user_preferences',
+    protected $defaultIncludes = [
+        'country',
+        'cover',
+        'is_bng',
+        'is_full_bn',
+        'is_gmt',
+        'is_limited_bn',
+        'is_moderator',
+        'is_nat',
+        'is_restricted',
     ];
 
+    protected $permissions = [
+        'friends' => 'IsNotOAuth',
+        'is_bng' => 'IsNotOAuth',
+        'is_full_bn' => 'IsNotOAuth',
+        'is_gmt' => 'IsNotOAuth',
+        'is_limited_bn' => 'IsNotOAuth',
+        'is_moderator' => 'IsNotOAuth',
+        'is_nat' => 'IsNotOAuth',
+        'is_restricted' => 'IsNotOAuth',
+    ];
+
+    public function __construct()
+    {
+        static $includes;
```

Comment:
looking again, I wonder if it's better to define all the includes in `UserCompact` instead.

Reply:
that does sound more useful 🤔